scene/2d/tile_map_layer_group.cpp

```diff
@@ -110,7 +110,7 @@ void TileMapLayerGroup::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_tileset", "tileset"), &TileMapLayerGroup::set_tileset);
 	ClassDB::bind_method(D_METHOD("get_tileset"), &TileMapLayerGroup::get_tileset);
 
-	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "tile_set", PROPERTY_HINT_RESOURCE_TYPE, "TileSet"), "set_tileset", "get_tileset");
+	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "tile_set", PROPERTY_HINT_RESOURCE_TYPE, "TileSet", PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_EDITOR_INSTANTIATE_OBJECT), "set_tileset", "get_tileset");
```

Comment:
See #80038
TileMaps are usually used with a TileSet saved to file and shared between multiple scenes. The automatically created one is built-in, which makes it specific to the scene and also risks bloating the scene. By creating TileSet by default you are pointing users towards wrong usage of the node. It's like making Sprite2D with default empty texture.

This might apply to other resources, so you might want to re-check them with that in mind. It comes down to how the node is usually used.

Reply:
I've undone the change for TileMap and GridMap.